compose/ui/ui-graphics/src/skikoMain/kotlin/androidx/compose/ui/graphics/SkiaGraphicsContext.skiko.kt

```diff
@@ -19,19 +19,22 @@ package androidx.compose.ui.graphics
 import androidx.compose.runtime.snapshots.SnapshotStateObserver
 import androidx.compose.ui.InternalComposeUiApi
 import androidx.compose.ui.graphics.layer.GraphicsLayer
-import kotlin.js.JsName
 
-/**
- * Create a new [GraphicsContext].
- */
 @InternalComposeUiApi
-@JsName("createGraphicsContext")
-fun GraphicsContext(snapshotObserver: SnapshotStateObserver): GraphicsContext =
-    SkiaGraphicsContext(snapshotObserver)
+class SkiaGraphicsContext() : GraphicsContext {
+    private val snapshotObserver = SnapshotStateObserver { command ->
+        command()
+    }
+
+    init {
+        snapshotObserver.start()
+    }
+
+    fun dispose() {
+        snapshotObserver.stop()
+        snapshotObserver.clear()
+    }
 
-private class SkiaGraphicsContext(
-    private val snapshotObserver: SnapshotStateObserver
-) : GraphicsContext {
     override fun createGraphicsLayer(): GraphicsLayer {
         return GraphicsLayer(snapshotObserver)
```

Comment:
```suggestion
        snapshotObserver.clear(layer)
        return GraphicsLayer(snapshotObserver)
```
Otherwise we have a leak of `layer`

Reply:
Sorry, it should be in `releaseGraphicsLayer()` or better in `layer.release()`, as the layer itself only knows on which it called `observeReads`